tests/linux_bridge_test.go

```diff
@@ -0,0 +1,86 @@
+package nmstate_tests
+
+import (
+	"io/ioutil"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	yaml "github.com/ghodss/yaml"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate.io/v1"
+)
+
+func waitBridge(nodeName string, name string, mustExist bool) error {
+
+	return wait.Poll(5*time.Second, 50*time.Second, func() (bool, error) {
+		var err error
+		nns, err := nmstateNNSs.Get(nodeName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		_, exist := findInterfaceInfo("br1", nns.Status.CurrentState.Interfaces)
+		return exist == mustExist, nil
+	})
+
+}
+
+var _ = Describe("Linux Bridge", func() {
+	Context("Create at node01 br1 with port eth1", func() {
+		var (
+			expectedState *nmstatev1.NodeNetworkState
+
+			_ = BeforeEach(func() {
+				// TODO node01 is harcoded at linux-bridge.yaml
+				By("Creating bridge")
+				manifest, err := ioutil.ReadFile(*manifests + "create-br1-linux-bridge.yaml")
+				Expect(err).ToNot(HaveOccurred())
+
+				state := nmstatev1.NodeNetworkState{}
+				err = yaml.Unmarshal(manifest, &state)
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedState, err = defaultNNSs.Create(&state)
+				Expect(err).ToNot(HaveOccurred())
+
+			})
+
+			_ = AfterEach(func() {
+
+				// TODO node01 is harcoded at linux-bridge.yaml
+				By("Deleting bridge")
+				manifest, err := ioutil.ReadFile(*manifests + "delete-br1-linux-bridge.yaml")
+				Expect(err).ToNot(HaveOccurred())
+
+				nns := nmstatev1.NodeNetworkState{}
+				err = yaml.Unmarshal(manifest, &nns)
+				Expect(err).ToNot(HaveOccurred())
+
+				deleteBr1, err := defaultNNSs.Get(nodeName, metav1.GetOptions{})
+				deleteBr1.Spec = nns.Spec
+				defaultNNSs.Update(deleteBr1)
+
+				// Wait for bridge to be removed
+				waitBridge(nodeName, "br1", false)
+
+				writePodsLogs(GinkgoWriter)
+			})
+		)
+
+		It("should have bridge configured as expected", func() {
+			err := waitBridge(nodeName, "br1", true)
+			Expect(err).ToNot(HaveOccurred())
+			waitBridge(nodeName, "br1", true)
+			obtainedState, err := nmstateNNSs.Get(nodeName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			obtainedbr1, _ := findInterfaceInfo("br1", obtainedState.Status.CurrentState.Interfaces)
+			expectedbr1, _ := findInterfaceSpec("br1", expectedState.Spec.DesiredState.Interfaces)
+			Expect(obtainedbr1.InterfaceSpec.Bridge).To(Equal(expectedbr1.Bridge))
```

Comment:
this will not work once we don't pass all parameters (skip one and let the default value take place).

Reply:
nmstate completed the default ones so it should work, I think.